Upstream-Status: Backport

From 319d02609992273e887242ed9788db68d3310b6c Mon Sep 17 00:00:00 2001
From: "pixdamix@gmail.com"
 <pixdamix@gmail.com@e8e0d7a0-c8d9-11dd-a880-a1081c7ac358>
Date: Thu, 19 Jan 2012 13:52:06 +0000
Subject: [PATCH 01/10] Fixed opkg losing auto_installed flag on upgrading

Thanks-to: Christian Hitz <christian.hitz@aizo.com>

git-svn-id: http://opkg.googlecode.com/svn/trunk@635 e8e0d7a0-c8d9-11dd-a880-a1081c7ac358
---
 libopkg/opkg_install.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/libopkg/opkg_install.c b/libopkg/opkg_install.c
index 74a2ce1..3925f58 100644
--- a/libopkg/opkg_install.c
+++ b/libopkg/opkg_install.c
@@ -1449,6 +1449,9 @@ opkg_install_pkg(pkg_t *pkg, int from_upgrade)
                /* removing files from old package, to avoid ghost files */
                remove_data_files_and_list(old_pkg);
                remove_maintainer_scripts(old_pkg);
+
+               /* maintain the "Auto-Installed: yes" flag */
+               pkg->auto_installed = old_pkg->auto_installed;
 	  }
 
 
-- 
1.7.9.5

